core/src/main/java/org/elasticsearch/search/aggregations/metrics/avg/ParsedAvg.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.metrics.avg;
+
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.aggregations.metrics.ParsedNumericMetricsAggregation;
+
+import java.io.IOException;
+
+public class ParsedAvg extends ParsedNumericMetricsAggregation.SingleValue implements Avg {
+
+    @Override
+    public double getValue() {
+        return value();
+    }
+
+    @Override
+    protected String getType() {
+        return AvgAggregationBuilder.NAME;
+    }
+
+    private static final ObjectParser<ParsedAvg, Void> PARSER = new ObjectParser<>(AvgAggregationBuilder.NAME, true, ParsedAvg::new);
+
+    static {
+        declareCommonSingeValueFields(PARSER, Double.POSITIVE_INFINITY);
+    }
+
+    public static ParsedAvg fromXContent(XContentParser parser, final String name) {
+        ParsedAvg avg = PARSER.apply(parser, null);
+        avg.setName(name);
+        return avg;
+    }
+
+    @Override
+    protected XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
+        boolean hasValue = value != Double.POSITIVE_INFINITY;
```

Comment:
In InternalAvg the `value` field is only rendered if the avg normalizer (count) is not 0. (`count != 0 ? getValue() : null`). I choose to parse back `null` as Double.POSITIVE_INFINITY because that is what you get back when you divide a positive double by `0L`. So I check for that value here to get the same xContent output like InternalAvg.